llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -475,10 +481,9 @@ InstructionCost RISCVTTIImpl::getShuffleCost(TTI::ShuffleKind Kind,
 
       return LT.first *
              (TLI->getLMULCost(LT.second) + // FIXME: this should be 1 for andi
-              TLI->getLMULCost(
-                  LT.second) + // FIXME: vmv.x.s is the same as extractelement
               getRISCVInstructionCost({RISCV::VMV_V_I, RISCV::VMERGE_VIM,
-                                       RISCV::VMV_V_X, RISCV::VMSNE_VI},
```

Comment:
Hm, one thing I'm noticing.  I think the example code sequence above is only for short fixed vectors.  For that to work on scalable vectors, I think we need a slide in there somewhere.  Or, said alternatively, that remaining LMUL scaling is probably modeling the vslidedown more than the andi for scalable vectors and we need to adjust our example and comments slightly.  

(Not directly relevant to your change, more the existing code structure.)